modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1141,6 +1141,86 @@ def _apply_func_to_list_of_partitions(cls, func, partitions, **kwargs):
         preprocessed_func = cls.preprocess_func(func)
         return [obj.apply(preprocessed_func, **kwargs) for obj in partitions]
 
+    @classmethod
+    def to_single_partition(cls, partitions):
+        """
+        Call to_pandas for the given NumPy array of partitions.
+
+        For a NumPy array of partitions create a single partition
+        that has the data for all partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The partitions which have to be converted to a single partition.
+
+        Returns
+        -------
+        PandasDataframePartition
+            A PandasDataframePartition object which holds the data of all partitions.
```

Comment:
```suggestion
        np.ndarray
            A NumPy 2D array of a single partition.
```

Reply:
This function returns the output of partition.apply() which would just be one PandasDataframePartition this would be converted to numpy array in dataframe class.

Reply:
I see, but we need to change this to return a numpy array of partitions. The partition manager's methods should return a numpy array of new partitions so the dataframe could take it in the constrctor. We should not wrap partitions in a numpy array at the dataframe layer.

Reply:
okay